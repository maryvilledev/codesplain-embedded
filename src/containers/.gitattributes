# Denote yarn.lock as binary file
yarn.lock binary
