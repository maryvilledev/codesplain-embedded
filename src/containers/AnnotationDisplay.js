import React, { Component, PropTypes } from 'react';
import MarkdownRenderer from 'react-markdown-renderer';
import { connect } from 'react-redux';

import markdownRendererOptions from '../util/markdown-renderer-options';

const styles = {
  container: {
    flex: '1 1 auto',
  },
};

class AnnotationDisplay extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      annotations,
      selectedLine,
    } = this.props;
    if (selectedLine === -1) {
      return (
        <div style={styles.container}>
          <h1>Annotation</h1>
        </div>
      );
    }

    const annotation = annotations[selectedLine + 1];
    return (
      <div style={styles.container}>
        <h1>Annotation</h1>
        <MarkdownRenderer
          markdown={annotation.annotation}
          options={markdownRendererOptions}
        />
      </div>
    );
  }
}

AnnotationDisplay.propTypes = {
  selectedLine: PropTypes.number,
}

AnnotationDisplay.defaultProps = {
  selectedLine: -1,
}

const mapStateToProps = (state) => ({
  annotations: state.annotations,
  selectedLine: state.selectedLine,
});

export default connect(mapStateToProps)(AnnotationDisplay);
