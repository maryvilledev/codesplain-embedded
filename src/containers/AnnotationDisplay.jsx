import React, { Component, PropTypes } from 'react';
import MarkdownRenderer from 'react-markdown-renderer';
import { connect } from 'react-redux';

import { setSelectedLine } from '../actions/app';
import {
  getAnnotatedLines,
  getNextAnnotation,
  getPreviousAnnotation,
  hasNextAnnotation,
  hasPreviousAnnotation,
} from '../util/annotations';
import markdownRendererOptions from '../util/markdown-renderer-options';

const styles = {
  container: {
    flex: '1 1 auto',
    flexBasis: '50%', // "default width"
    paddingLeft: '15px',
  },
  markdownContainer: {
    background: '#fff',
    padding: '10px',
    borderRadius: '5px',
  },
  buttonContainer: {
    display: 'inline-flex',
    justifyContent: 'flex-start',
  },
};

class AnnotationDisplay extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hasPreceedingAnnotation: false,
      hasProceedingAnnotation: false,
    };
    this.getNextAnnotation = this.getNextAnnotation.bind(this);
    this.getPreviousAnnotation = this.getPreviousAnnotation.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    const {
      annotations,
      selectedLine,
    } = nextProps;
    const nextAnnotatedLines = getAnnotatedLines(annotations);
    this.setState({
      hasProceedingAnnotation: hasNextAnnotation(nextAnnotatedLines, selectedLine),
      hasPreceedingAnnotation: hasPreviousAnnotation(nextAnnotatedLines, selectedLine),
    });
  }

  getPreviousAnnotation() {
    const {
      annotations,
      dispatch,
      selectedLine,
    } = this.props;
    // Get the annotated lines
    const previous = getPreviousAnnotation(annotations, selectedLine);
    if (!previous) {
      return;
    }
    dispatch(setSelectedLine(previous.lineNumber));
  }

  getNextAnnotation() {
    const {
      annotations,
      dispatch,
      selectedLine,
    } = this.props;
    // Get the annotated lines
    const next = getNextAnnotation(annotations, selectedLine);
    if (!next) {
      return;
    }
    dispatch(setSelectedLine(next.lineNumber));
  }

  render() {
    const {
      hasPreceedingAnnotation,
      hasProceedingAnnotation,
    } = this.state;
    const {
      annotations,
      selectedLine,
    } = this.props;
    if (selectedLine === -1) {
      return (
        <div style={styles.container}>
          <h2>Annotation</h2>
        </div>
      );
    }
    const annotation = annotations[selectedLine];
    return (
      <div style={styles.container}>
        <h2>Annotation</h2>
        <div style={styles.buttonContainer}>
          <button
            disabled={hasPreceedingAnnotation}
            onClick={this.getPreviousAnnotation}
            type="button"
          >
            Previous
          </button>
          <button
            disabled={hasProceedingAnnotation}
            onClick={this.getNextAnnotation}
            type="button"
          >
            Next
          </button>
        </div>
        <div style={styles.markdownContainer}>
          <MarkdownRenderer
            markdown={annotation.annotation}
            options={markdownRendererOptions}
          />
        </div>
      </div>
    );
  }
}

AnnotationDisplay.propTypes = {
  selectedLine: PropTypes.number,
}

AnnotationDisplay.defaultProps = {
  selectedLine: -1,
}

const mapStateToProps = (state) => ({
  annotations: state.annotations,
  selectedLine: state.selectedLine,
});

export default connect(mapStateToProps)(AnnotationDisplay);
