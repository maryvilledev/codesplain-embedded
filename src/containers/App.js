import React, {Component, PropTypes} from 'react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

const mockReducer = (state={}, action) => state

class App extends Component {
  constructor(props) {
    super(props);
    this.store = createStore(mockReducer);
  }
  render() {
    return (
      <Provider store={this.store}>
        <div>Ello govna?</div>
      </Provider>
    );
  }
}

App.propTypes = {
  snippetKey: PropTypes.string.required,
}

export default App
