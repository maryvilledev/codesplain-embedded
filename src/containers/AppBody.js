import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import axios from 'axios';

class AppBody extends Component {
  componentDidMount() {
    const { dispatch, snippetKey } = this.props;
    axios.get("../public/snippet.json")
      .then((res) => {
        console.log(res.data.snippetTitle)
      });
  }
  render() {
    return null;
  }
}

AppBody.propTypes = {
  snippetKey: PropTypes.string.required,
}

export default connect()(AppBody);
