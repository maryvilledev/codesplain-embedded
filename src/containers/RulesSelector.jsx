import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import RuleLabel from '../components/RuleLabel';
import { toggleRule } from '../actions/app';

const styles = {
  container: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'center',
  },
};

const makeListItems = (dispatchToggleRule, filters) => (
  Object.keys(filters)
    .map((filterName) => {
      const {
        color,
        count,
        prettyTokenName,
        selected,
      } = filters[filterName];
      return (
        <RuleLabel
          color={color}
          count={count}
          key={filterName}
          onClick={dispatchToggleRule}
          rule={prettyTokenName}
          selected={selected}
          value={filterName}
        />
      )
    })
)

const RulesSelector = ({ dispatchToggleRule, filters }) => {
  const listItems = makeListItems(dispatchToggleRule, filters);
  return (
    <div style={styles.container}>
      {listItems}
    </div>
  );
}

RulesSelector.propTypes = {
  dispatchToggleRule: PropTypes.func.isRequired,
  filters: PropTypes.shape({
    color: PropTypes.string,
    count: PropTypes.number,
    prettyTokenName: PropTypes.string,
    selected: PropTypes.bool,
  }).isRequired,
};

const mapStateToProps = (state) => {
  const {
    filters,
  } = state;
  return {
    filters,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    dispatchToggleRule: (rule) => { dispatch(toggleRule(rule)); },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(RulesSelector);
