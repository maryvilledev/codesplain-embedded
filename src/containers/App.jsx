import React, {Component, PropTypes} from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import AppBody from './AppBody';
import appReducer from '../reducers/app';

class App extends Component {
  constructor(props) {
    super(props);
    this.store = createStore(appReducer);
  }
  render() {
    return (
      <Provider store={this.store}>
        <AppBody {...this.props} />
      </Provider>
    );
  }
}

export default App;
