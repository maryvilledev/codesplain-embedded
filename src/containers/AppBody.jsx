import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import axios from 'axios';
import { setState } from '../actions/app';

import AnnotationDisplay from './AnnotationDisplay';
import RulesSelector from '../components/RulesSelector';
import SnippetArea from './SnippetArea';
import Title from './Title';

const API_URL = process.env.API_URL;
const styles = {
  container: {
    display: 'flex',
    flexFlow: 'row wrap',
    justifyContent: 'space-evenly',
  },
};

class AppBody extends Component {
  componentDidMount() {
    const { dispatch, snippetKey } = this.props;
    const [user, snippet] = snippetKey.split('/');
    axios.get(`${API_URL}/users/${user}/snippets/${snippet}`)
      .then((res) => {
        dispatch(setState(res.data));
      });
  }
  render() {
    return (
      <div>
        <Title />
        <RulesSelector />
        <div style={styles.container}>
          <SnippetArea />
          <AnnotationDisplay />
        </div>
      </div>
    );
  }
}

AppBody.propTypes = {
  snippetKey: PropTypes.string.isRequired,
}

export default connect()(AppBody);
