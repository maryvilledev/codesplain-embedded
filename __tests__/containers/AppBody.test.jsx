import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

import { AppBody } from '../../src/containers/AppBody';

describe('<AppBody />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <AppBody
        snippetKey="rick-sanchez/formula_for_concentrated_dark_matter"
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
