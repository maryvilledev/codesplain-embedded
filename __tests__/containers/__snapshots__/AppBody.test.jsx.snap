// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBody /> matches snapshot 1`] = `
<div
  style={
    Object {
      "background": "#f5f6f5",
      "borderRadius": "5px",
      "margin": "15px",
      "padding": "10px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <Connect(Title)
      snippetKey="rick-sanchez/formula_for_concentrated_dark_matter"
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <Connect(RulesSelector) />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexFlow": "row wrap",
        "justifyContent": "space-evenly",
      }
    }
  >
    <Connect(SnippetArea) />
    <Connect(AnnotationDisplay) />
  </div>
</div>
`;
